Product api endpoints

Omarjjj opened the pull request.

Omarjjj committed:
feat: implement product API endpoints

- Add GET /api/products/totals - get total amount for all products
- Add GET /api/products - get product details with amounts
- Add GET /api/products/:id/total - get product by ID with total amount
- Add GET /api/products/:id/amounts - get product by ID with all amounts
- Add POST /api/products - add new product entry
- Simplify code structure and improve error handling
- Add data persistence to JSON files

Omarjjj committed:
creating an api

Omarjjj committed:
update api implementation

Omarjjj committed:
clean up files

Omarjjj commented:
API implementation

hani1992 commented on dist/index.js:
dist files shouldn’t be committed. Please add this to .gitignore

hani1992 commented on dist/server.js:
dist files shouldn’t be committed. Please add this to .gitignore

hani1992 commented on src/server.ts:
```diff
@@ -1,12 +1,166 @@
+});
+
+// get product details (product IDs with their amounts)
+app.get('/api/products', async (req, res) => {
```
return a simple list of all transactions

Omarjjj committed:
clearing up files

hani1992 commented on src/server.ts:
```diff
@@ -1,12 +1,166 @@
+//  add new product
+app.post('/api/products', async (req, res) => {
+    try {
+        const { productId, amount } = req.body;
```
Add a check for the existence of `req.body` and return an appropriate error response if it's missing.

Omarjjj committed:
Fix TypeScript errors and update server configuration

- Fixed validateProduct function signature and validation logic
- Added missing helper functions (handleError, sendResponse)
- Fixed property access error for totals[id]
- Added express.json() middleware for request body parsing
- Changed server port to 3001 to avoid port conflicts
- Added console logging for server startup
- Updated error handling in POST endpoint

Omarjjj committed:
adding middelware , middelware auth , configure& connecting it to a mongoDB , and performe CRUD operaion
